x-pack/plugins/alerting/server/authorization/alerting_authorization_kuery.test.ts

```diff
@@ -155,332 +155,799 @@ describe('asKqlFiltersByRuleTypeAndConsumer', () => {
-              myAppWithSubFeature: { read: true, all: true },
+            type: AlertingAuthorizationFilterType.ESDSL,
+            fieldNames: {
+              ruleTypeId: 'path.to.rule.id',
+              consumer: 'consumer-field',
+              spaceIds: 'path.to.space.ids',
             },
-            enabledInLicense: true,
           },
+          'space1'
+        )
+      ).toEqual({
+        bool: {
+          filter: [
+            {
+              bool: {
+                minimum_should_match: 1,
+                should: [{ match: { 'path.to.rule.id': 'myAppAlertType' } }],
+              },
+            },
+            { bool: { filter: [{ term: { 'path.to.space.ids': 'space1' } }] } },
+            {
+              bool: { minimum_should_match: 1, should: [{ match: { 'consumer-field': 'myApp' } }] },
+            },
+          ],
+        },
+      });
+    });
+
+    test('constructs ES DSL filter for single rule type with multiple authorized consumers', async () => {
+      expect(
+        asFiltersByRuleTypeAndConsumer(
+          new Set([
+            {
+              actionGroups: [],
+              defaultActionGroupId: 'default',
+              minimumLicenseRequired: 'basic',
+              isExportable: true,
+              recoveryActionGroup: RecoveredActionGroup,
+              id: 'myAppAlertType',
+              name: 'myAppAlertType',
+              producer: 'myApp',
+              authorizedConsumers: {
+                alerts: { read: true, all: true },
+                myApp: { read: true, all: true },
+                myOtherApp: { read: true, all: true },
+              },
+              enabledInLicense: true,
+            },
+          ]),
           {
-            actionGroups: [],
-            defaultActionGroupId: 'default',
-            minimumLicenseRequired: 'basic',
-            isExportable: true,
-            recoveryActionGroup: RecoveredActionGroup,
-            id: 'myOtherAppAlertType',
-            name: 'myOtherAppAlertType',
-            producer: 'alerts',
-            authorizedConsumers: {
-              alerts: { read: true, all: true },
-              myApp: { read: true, all: true },
-              myOtherApp: { read: true, all: true },
-              myAppWithSubFeature: { read: true, all: true },
+            type: AlertingAuthorizationFilterType.ESDSL,
+            fieldNames: {
+              ruleTypeId: 'path.to.rule.id',
+              consumer: 'consumer-field',
+              spaceIds: 'path.to.space.ids',
             },
-            enabledInLicense: true,
           },
+          'space1'
+        )
+      ).toEqual({
+        bool: {
+          filter: [
+            {
+              bool: {
+                minimum_should_match: 1,
+                should: [{ match: { 'path.to.rule.id': 'myAppAlertType' } }],
+              },
+            },
+            { bool: { filter: [{ term: { 'path.to.space.ids': 'space1' } }] } },
```

Comment:
Does this have to be `{ bool: { filter: [{ term: {...`? I feel like it is simpler to just have `{ term: { 'path.to.space.ids': 'space1' } }`?

Reply:
Updated!